src/main/cpp/threadutility.cpp

```diff
@@ -0,0 +1,91 @@
+#include "log4cxx/helpers/threadutility.h"
+#include "log4cxx/private/log4cxx_private.h"
+#include "log4cxx/helpers/loglog.h"
+
+#include <signal.h>
+
+#if LOG4CXX_HAS_SETTHREADDESCRIPTION
+#include <windows.h>
+#include <processthreadsapi.h>
+#endif
+
+using log4cxx::helpers::ThreadUtility;
+
+struct ThreadUtility::priv_data{
```

Comment:
This is related to [LOGCXX-516](https://issues.apache.org/jira/browse/LOGCXX-516), to make things ABI-stable.  The CPP Core Guidelines suggest using `impl` as the class name, Qt uses `d_ptr`, I like using `priv_data`.  Effectively all this is doing is putting all of the instance variables into a hidden struct so that any users of the code don't need to be recompiled incase we add/remove members into the class.

I don't have a strong opinion as to what we name it.

Reply:
All of those phrases are named pretty meaningless in the end. But your `priv_data` seems more descriptive than `impl`, so let's keep that.